ubuntu22.04/nvidia-driver

```diff
@@ -480,12 +482,32 @@ _shutdown() {
 }
 
 _resolve_kernel_type() {
+
+  # For backwards compatibility with older GPU Operator versions where KERNEL_MODULE_TYPE is not set,
+  # honor the deprecated OPEN_KERNEL_MODULES_ENABLED field
+  if [ -z "${KERNEL_MODULE_TYPE}" ]; then
+   if [ ${OPEN_KERNEL_MODULES_ENABLED} == "true" ]; then
+     echo "Honoring deprecated field OPEN_KERNEL_MODULES_ENABLED=${OPEN_KERNEL_MODULES_ENABLED}"
+     KERNEL_TYPE=kernel-open
+     return 0
+   fi
+
+   # If neither KERNEL_MODULE_TYPE or OPEN_KERNEL_MODULES_ENABLED are set, get the default
+   # kernel module type from the nvidia-installer.
+   KERNEL_TYPE=$(nvidia-installer --print-recommended-kernel-module-type)
```

Comment:
note that this command `nvidia-installer --print-recommended-kernel-module-type` returns either `open` or `proprietary`. 

i) If it's `open`, we set `KERNEL_TYPE=kernel-open`
ii) If it's `proprietary`, we set `KERNEL_TYPE=kernel`

Reply:
Thanks for pointing this out. Will update this in my next iteration.

Reply:
Actually, I will remove this call altogether as we agreed to maintain the old behavior, we would default to proprietary in this case.